File: python/infinicore/ops/broadcast_to.py
from infinicore.lib import _infinicore
from infinicore.tensor import Tensor


# 修改说明：将参数名 'shape' 改为 'size'，以匹配测试用例中的调用方式 kwargs={size=...}
def broadcast_to(input, size, *, out=None):
    if out is None:
        return Tensor(_infinicore.broadcast_to(input._underlying, size))

    _infinicore.broadcast_to_(out._underlying, input._underlying)

    return out

File: csrc/infinicore/src/ops/broadcast_to/broadcast_to.cc
#include "infinicore/ops/broadcast_to.hpp"
#include "../../utils.hpp"

namespace infinicore::op {

INFINICORE_GRAPH_OP_DISPATCHERS_IMPL(BroadcastTo);

BroadcastTo::BroadcastTo(Tensor y, Tensor x) {
    INFINICORE_ASSERT_TENSORS_SAME_DEVICE(y, x);
    INFINICORE_GRAPH_OP_DISPATCH(y->device().type(), y, x);
}

void BroadcastTo::execute(Tensor y, Tensor x) {
    INFINICORE_GRAPH_OP_RECORD_OR_RUN(BroadcastTo, y, x);
}

Tensor broadcast_to(Tensor x, const std::vector<int64_t> &shape) {
    Shape target_shape(shape.begin(), shape.end());
    auto y = Tensor::empty(target_shape, x->dtype(), x->device());
    broadcast_to_(y, x);
    return y;
}

void broadcast_to_(Tensor y, Tensor x) {
    BroadcastTo::execute(y, x);
}

} // namespace infinicore::op
